torch/csrc/autograd/profiler.h

```diff
@@ -24,16 +29,93 @@ constexpr inline std::size_t ceilToMultiple(std::size_t a, std::size_t b) {
   return ((a + b - 1) / b) * b;
 }
 
+inline uint64_t getTime() {
+  using namespace std::chrono;
+  using clock = std::conditional<high_resolution_clock::is_steady, high_resolution_clock, steady_clock>::type;
+  return duration_cast<nanoseconds>(clock::now().time_since_epoch()).count();
+}
+
 enum class EventKind {
   Mark,
   PushRange,
   PopRange
 };
+#ifdef WITH_CUDA
+static void cudaCheck(cudaError_t result, const char * file, int line) {
+  if(result != cudaSuccess) {
+    std::stringstream ss;
+    ss << file << ":" << line << ": " << cudaGetErrorString(result);
+    throw std::runtime_error(ss.str());
+  }
+}
+#define PROFILER_CUDA_CHECK(result) cudaCheck(result,__FILE__,__LINE__);
+#endif
```

Comment:
Can we have a single macro for this? There are equivalent definitions sprinkled across the codebase. Can you move it to `torch/csrc/Exceptions.h`?